Source/common/SNTConfigurator.m

```diff
@@ -101,7 +103,8 @@ - (instancetype)init {
         kMachineOwnerKey, kMachineIDKey, kMachineOwnerPlistFileKey, kMachineOwnerPlistKeyKey,
         kMachineIDPlistFileKey, kMachineIDPlistKeyKey
     ];
-    [self reloadConfigData];
+    _reloading = 0;
+    [self reloadConfigDataHelper];
```

Comment:
This should call reloadConfigData so the barrier can be alive even during init.

Reply:
I avoided doing that because of the 5 sec blocking time. Every component of Santa initializes an instance of SNTConfigurator very early on and will block until complete:

[2018-02-02T00:52:41.882Z] I santad: Started, version 0.9.21
[2018-02-02T00:52:47.118Z] I santad: Connected to driver, activating.

⇒  date && santactl status && date
Thu Feb  1 19:57:05 EST 2018
>>> Daemon Info
  Mode                      | Lockdown
  File Logging              | Yes
  Watchdog CPU Events       | 0  (Peak: 0.22%)
  Watchdog RAM Events       | 0  (Peak: 12.68MB)
...
Thu Feb  1 19:57:10 EST 2018

Reply:
GitHub didn't save my updated comment - yeah the 5s sleep needs to be avoided but you still need the barrier. Either do the atomic stuff around the call in init or make the sleep length a parameter

Reply:
ack